src/QsCompiler/QirGeneration/README.md

```diff
@@ -0,0 +1,64 @@
+# QIR Emission - Preview Feature
+
+This file contains directions for using the preview feature integrated into the Q# compiler to emit QIR.
+QIR is a convention for how to represent quantum programs in LLVM. Its specification can be found [here](https://github.com/microsoft/qsharp-language/tree/main/Specifications/QIR#quantum-intermediate-representation-qir). 
+We aim to ultimately move the Q# compiler to be fully LLVM-based. While the emission is supported starting with the March 2021 release, it is as of this time not yet connected to the runtime. The QIR runtime and instructions for how to execute the emitted QIR can be found [here](https://github.com/microsoft/qsharp-runtime/tree/main/src/QirRuntime#the-native-qir-runtime). We are working on a full integration in the future.
+
+## Using the Q# Compiler to Emit QIR
+
+QIR can be emitted for any Q# project as long as its output type is an executable.
+To enable QIR emission, open the project file in a text editor and add the following project property:
+```
+<QirGeneration>true</QirGeneration>
+```
+If the project builds successfully, the .ll file containing QIR can be found in `qir` folder in the project folder. Alternatively, the folder path can be specified via the `QirOutputPath` project property. The project file should look similar to this:
```

Comment:
Very minor: right now we are actually producing both the human readable .ll file and the more tightly packed .bc (bitcode) file. I'm not sure if this is worth calling out in the readme, but we may eventually want to generate just one by default and have the the other as an option, so perhaps introducing the concepts now would be good.

Reply:
Hm, I am not sure what you are referring to; I believe we are currently only generating .ll, though I agree that we should be generating bitcode in the near future, possibly with an option to keep with human readable.

Reply:
Hmm... the update to also produce bitcode was in the feature branch but seems to have been lost in a merge somewhere. Here's the commit that introduced it: https://github.com/microsoft/qsharp-compiler/commit/722ec70a97b65f8d3ee1085368142a91183969db#diff-d72a80363d839b02d5bf562da75e9641615d62dce924b53e85790cde9d1d8179. That commit is still in the history, but the code isn't in the file anymore, and I can't seem to find the commit that removed it. Was that intentionally dropped somewhere along the way?

Reply:
Ah, I found it. #889 included enough updates to the file Context.cs that the diff registered it as an entire deletion of the existing `Emit` function and replacing with a new `Emit` function. As a result, the merge automatically lost the change to produce bitcode. It's not a big deal, since as the diff I linked above shows it's extremely easy to produce, so we can add it back in when we are ready. I double checked the various repos and no test infrastructure is taking any dependencies on the bitcode being present, so there is no need to introduce any more churn to the feature branch while we are trying to merge to main.